src/hotspot/share/opto/addnode.cpp

```diff
@@ -786,6 +845,51 @@ Node* OrINode::Ideal(PhaseGVN* phase, bool can_reshape) {
        return new RotateRightNode(in(1)->in(1), shift, TypeInt::INT);
      }
   }
+  // major_progress() check postpones the transformation after loop optimization
+  if (can_reshape && !phase->C->major_progress() && Matcher::match_rule_supported(Op_BitfieldInsertI)) {
```

Comment:
There's `C->post_loop_opts_phase()` flag added recently specifically for that purpose.